### PR TITLE
Twu proposal view/edit/delete

### PR DESCRIPTION
**Pull request by bdolor**
This PR closes issue: [issue DM-1171]

Includes tests? [N]
Updated docs? [Y]

Proposed changes:
- adds dashboard view and fixes readOne proposal view
- adds editing capabilities to proposals
- adds delete

### CODE REVIEW

File: src/back-end/lib/permissions.ts [hunk 1]
```diff
@@ -740,6 +741,50 @@ export async function editTWUOpportunity(
   );
 }
 
+/**
```
Comment (dhruvio) [inline src/back-end/lib/permissions.ts:744]: Add `createDraftTWUProposal`.

File: src/back-end/lib/permissions.ts [hunk 2]
```diff
@@ -740,6 +741,50 @@ export async function editTWUOpportunity(
...
+export async function submitTWUProposal(
+  connection: Connection,
+  session: Session,
+  proposalId: string
+): Promise<boolean> {
+  return (
+    (session &&
+      (await isTWUProposalAuthor(connection, session.user, proposalId)) &&
+      (await hasAcceptedCurrentTerms(connection, session))) ||
+    false
```
Comment (dhruvio) [inline src/back-end/lib/permissions.ts:784]: Do an organization check for correct TWU qualification.

File: src/back-end/lib/db/proposal/team-with-us.ts [hunk 3]
```diff
@@ -521,6 +522,26 @@ export const readManyProposalResourceQuestionResponses = tryDb<
   return valid(results);
 });
 
+/**
+ * Fetches members from the db, associated with a Proposal
+ */
+export const readTWUProposalMembers = tryDb<[Id], RawProposalTeamMember[]>(
```
Comment (dhruvio) [inline src/back-end/lib/db/proposal/team-with-us.ts:528]: Remove `export` keyword

File: src/back-end/lib/db/proposal/team-with-us.ts [hunk 4]
```diff
@@ -587,6 +608,19 @@ export const readOneTWUProposal = tryDb<
     }
   }
 
+  // Read the members associated with the proposal from the db
+  const teamMembers =
+    getValidValue(await readTWUProposalMembers(connection, result.id), []) ??
+    [];
```
Comment (dhruvio) [inline src/back-end/lib/db/proposal/team-with-us.ts:614]: Remove `?? []`

Comment (bdolor) [reply src/back-end/lib/db/proposal/team-with-us.ts:614]: though, removing it generates a `teamMembers is possibly null` error further down in the code.

File: src/back-end/lib/permissions.ts (hunk 2 above)
Comment (bdolor) [reply src/back-end/lib/permissions.ts:784]: leaving for Ian in another PR

File: src/back-end/lib/permissions.ts (hunk 1 above)
Comment (bdolor) [reply src/back-end/lib/permissions.ts:744]: leaving for Ian

File: src/back-end/lib/db/proposal/team-with-us.ts (hunk 3 above)
Comment (bdolor) [reply src/back-end/lib/db/proposal/team-with-us.ts]: done

File: src/back-end/lib/resources/proposal/team-with-us.ts [hunk 5]
```diff
@@ -359,7 +345,12 @@ const create: crud.Create<
           organization: organization || undefined,
           status: validatedStatus.value,
           attachments: validatedAttachments.value,
-          team
+          team: team
+            ? team.map((t) => ({
+                member: getString(t, "member"),
+                hourlyRate: getNumber(t, "hourlyRate")
+              }))
+            : []
```
[the remark this answers is not in the document]
Comment (bdolor) [reply src/back-end/lib/resources/proposal/team-with-us.ts]: done

File: src/back-end/lib/db/proposal/team-with-us.ts (hunk 4 above)
Comment (IanFonzie) [reply src/back-end/lib/db/proposal/team-with-us.ts:614]: K easier to leave it then

**Review by IanFonzie — APPROVED** [review]

**Pull request merged**